src/Rule/Uuid.php

```diff
@@ -23,17 +21,41 @@ class Uuid extends Regex
     const INVALID_UUID = 'Uuid::INVALID_UUID';
 
     /**
-     * UUID Version 4.
+     * UUID NIL & version binary masks
      */
-    const UUID_V4 = 4;
+    const UUID_NIL = 0b000001;
+    const UUID_V1  = 0b000010;
+    const UUID_V2  = 0b000100;
+    const UUID_V3  = 0b001000;
+    const UUID_V4  = 0b010000;
```

Comment:
The reason you're doing it this way is making more sense to me now :). Especially with being able to specify multiple UUID versions numbers, which looks great. Thank you for your detailed explanation! I played around with the code a bit which is looking good too. I'm leaning towards a :+1: Now, lets just hope that no one actually provided 4 as an value. I'm submitting some small suggestions to make it more clear how to correctly implement it when it was done in a wrong way.